src/controllers/controller.js

```diff
@@ -1,141 +1,31 @@
-            documentType: req.body.documentType,
-            documentNumber: req.body.documentNumber,
-            owner:{
-                fullName: req.body.ownerName,
-                phoneNumber: req.body.ownerPhoneNumber,
-                email: req.body.ownerEmail
-            },
-            whoFound:{
-                fullName: req.body.foundName,
-                phoneNumber: req.body.foundPhoneNumber,
-                email: req.body.foundEmail
-            },
-            status: {
-                isLost: true,
-            },
-            location: {
-                lostPlace: req.body.location
-            },
-            requireReward: req.body.reward
-        });
-
-        //checking if the lost documents has found before
-        const theFound = await LostDocuments.findOne({
-            $and: [
-                {documentNumber: req.body.documentNumber},
-                {documentType: req.body.documentType},
-                {'status.isFound':true}
-            ]
-        });
-
-        //if the document is found before or to save a lost document
-        if (theFound) {
-            try {
-                const update = await LostDocuments.updateOne({_id: theFound._id}, 
-                    {$set: {'status.isLost':true, 'location.lostPlace':req.body.location}});
-    
-                res.json({
-                    msg: `Your ${theFound.documentType} has found by:`,
-                    whoFoundIt: {
-                        Name: theFound.whoFound.fullName,
-                        phoneNumber: theFound.whoFound.phoneNumber,
-                        email: theFound.whoFound.email
-                    }
-    
-                }); 
-            } catch (error) {
-                res.json(error.message);
-            } 
-        }
-        else{
-            try {
-                const lostDoc = await lost.save();
-                res.json(lostDoc);
-            } catch (error) {
-                res.json(error.message);
-            }
-        };
-    };
-
-    static async lostAndfoundDoc (req, res) {
-        const lost_found = await LostDocuments.find({
-            $and: [
-                {'status.isLost':true},
-                {'status.isFound':true}
-            ]
-        })
-        .select({_id:0, __v:0, status:0});
-        try {
-            res.json(lost_found)
-        } catch (error) {
-            res.json(error.message)
-        }
-    }   
-};
-
-export default Controller;
\ No newline at end of file
+import {Router} from 'express';
+import Lost from '../../models/model';
+
+const router = new Router();
+
+router.post('/lost', async (req, res)=>{
+
+    const document = await Lost.findOne({name: req.body.name});
+    if(document) return res.status(400).json({msg: 'document exist'});
+
+    const lost = new Lost({
+        name: req.body.name,
+        number: req.body.number
+    });
+    try {
+        const savedDoc  = await lost.save();
+        res.json({msg: "successfully saved"})
```

Comment:
Actually the record created should also be returned.

Reply:
In addition, when I create other records it keeps responding document exist, not sure if that is the intended behavior.